src/main/java/core/basesyntax/model/impl/ReporterImpl.java

```
@@ -0,0 +1,14 @@
+package core.basesyntax.model.impl;
+
+import core.basesyntax.db.Database;
+import core.basesyntax.model.Report;
+import core.basesyntax.model.Reporter;
+
+public class ReporterImpl implements Reporter {
+    @Override
+    public Report reportWith(Database database) {
+        Report.Builder builder = Report.builder();
+        database.consume(builder::set);
```

`reportWith(Database database)` does not validate the `database` parameter. The project explicitly requires handling null parameters ("Null parameters"). Add an explicit null check and throw an informative exception (fail fast), for example: `Objects.requireNonNull(database, "Database must not be null")` or throw a `RuntimeException` with a clear message. This will make errors in tests and at runtime actionable. See the project guidance that lists "Null parameters" among invalid inputs to anticipate.